### PR TITLE
Add `/trackmenu` command

### PR DESCRIPTION
**Pull request by riidefi**

### CODE REVIEW

**Comment by riidefi** [comment]
Renamed as "Add `/trackmenu` command." Depends on #810.

**Review by GnomedDev — CHANGES_REQUESTED** [review]
As said on discord, since this is a debugging option that is already using the commands system, it should just be a simple command of `/tracktest path slot music`. This menu is out of place and a maintenance burden.

**Comment by riidefi** [comment]
Thank you for sharing your perspective on this matter, and I genuinely appreciate the time and thought you put into it. However, in the current context, our focus is not oriented towards integrating such functionality into the command system. While your idea does have its merits, it requires a broader discussion and more substantial development implications than we are prepared to address at this point in time. I hope you understand our position.

I believe the code in my PR is comprehensive and robust, ready to withstand rigorous examination. It's already valuable to anyone who is testing tracks. Far from posing a maintenance challenge, the code operates in an optimal manner, effectively addressing all intended requirements.

**Comment by vabold** [comment]
Please use LayoutUIControls and other existing infrastructure to handle selection.

**Comment by riidefi** [comment]
> Please use LayoutUIControls and other existing infrastructure to handle selection.

Fixed in commit bcf2403

**Comment by riidefi** [comment]
Should be reviewable commit-by-commit.

File: payload/game/ui/page/CourseDebugPage.cc [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+    sentinel end() {
+        return {};
+    }
+
+private:
+    DirHandle m_dirHandle;
+};
+
+static_assert(std::ranges::input_range<DirEntryRange>,
+        "DirEntryRange is not an STL-compatible input range");
```
Comment (GnomedDev) [inline payload/game/ui/page/CourseDebugPage.cc:100]: This should be split off into a separate PR, and kept in it's own header.

File: assets/control/anim/CourseDebug.brlan.json5 [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                                    ],
+                                },
+                            ],
+                        },
+                    ],
+                },
+            ],
+        },
+    ],
+}
```
Comment (GnomedDev) [inline assets/control/anim/CourseDebug.brlan.json5:111]: Newline either here, or remove the one at the end of the brlyt.

**Review by GnomedDev — CHANGES_REQUESTED** [review]

File: payload/game/ui/page/CourseDebugPage.cc (hunk 1 above)
Comment (riidefi) [reply payload/game/ui/page/CourseDebugPage.cc]: Sure

**Comment by GnomedDev** [comment]
This should be set as draft until it is rebased on-top of SP after #827 is merged.

**Comment by riidefi** [comment]
> This should be set as draft until it is rebased on-top of SP after https://github.com/stblr/mkw-sp/pull/827 is merged.

I appreciate your concern, but Github automatically does that so this is pointless. If stebler merges 827, and then this PR, commit 707812 will be omitted from the history.

**Comment by riidefi** [comment]
Rebased on top of Gnome's work.

**Comment by GnomedDev** [comment]
This should not have been merged without anyone's positive review.

**Pull request merged**